CMakeLists.txt

```diff
@@ -158,13 +158,29 @@ if (XSDK_ENABLE_Fortran)
    enable_language(Fortran)
 endif()
 
-# Suppress warnings about MACOSX_RPATH
-set(CMAKE_MACOSX_RPATH OFF CACHE BOOL "")
+set(CMAKE_MACOSX_RPATH ON CACHE BOOL "")
 
 # CMake needs to know where to find things
 set(MFEM_CMAKE_PATH ${PROJECT_SOURCE_DIR}/config)
 set(CMAKE_MODULE_PATH ${MFEM_CMAKE_PATH}/cmake/modules)
 
+# Load standard install paths (but keep lib as the default on all platforms)
+set(CMAKE_INSTALL_LIBDIR lib CACHE PATH "Object code libraries (lib)")
+include(GNUInstallDirs)
+
+# Use RPATH on MacOS to avoid re-linking which invalidates code signatures
+file(RELATIVE_PATH REL_LIB_DIR "${CMAKE_INSTALL_FULL_BINDIR}" "${CMAKE_INSTALL_FULL_LIBDIR}")
+if (APPLE)
```

Comment:
I don't quite follow -- what issue is this solving?

Reply:
On Mac arm64, there are tons of warnings during the install phase, e.g.
```
-- Installing: /Users/lindad/projects/mfem-install-miniapp-content/installed/examples/ex27
/Users/lindad/miniforge/envs/moose-mpich/bin/install_name_tool: warning: changes being made to the file will invalidate the code signature in: /Users/lindad/projects/mfem-install-miniapp-content/installed/examples/ex27
[cctools-port]: generating fake signature for '/Users/lindad/projects/mfem-install-miniapp-content/installed/examples/ex27'
/Users/lindad/miniforge/envs/moose-mpich/bin/install_name_tool: warning: changes being made to the file will invalidate the code signature in: /Users/lindad/projects/mfem-install-miniapp-content/installed/examples/ex27
[cctools-port]: generating fake signature for '/Users/lindad/projects/mfem-install-miniapp-content/installed/examples/ex27'
```
By using the install rpath for the build binaries, then I could avoid re-linking and avoid these warnings ... of course this would have consequences for `make check`

Reply:
However, motivated by @nmnobre digging into this with his own Mac ARM machine and not seeing any warnings, I built an entire stack from scratch using my Mac's system tools instead of conda tools ... and then I also saw no warnings. So conda seems to be the foul player here. I will revert this work